src/main/g8/src/main/scala/$package$/api/protocol.scala

```diff
@@ -2,20 +2,36 @@ package $package$.api
 
 import $package$.api.healthcheck.DbStatus
 import $package$.domain.{ Item, ItemId }
-import de.heikoseeberger.akkahttpplayjson.PlayJsonSupport
-import play.api.libs.json.Json
+import zio.json._
 
 final case class CreateItemRequest(name: String, price: BigDecimal)
 final case class UpdateItemRequest(name: String, price: BigDecimal)
 final case class PartialUpdateItemRequest(name: Option[String], price: Option[BigDecimal])
+final case class EmptyResponse()
 
-trait JsonSupport extends PlayJsonSupport {
-  implicit val itemIdFormat                   = Json.valueFormat[ItemId]
-  implicit val itemFormat                     = Json.format[Item]
-  implicit val dbStatusFormat                 = Json.format[DbStatus]
-  implicit val createItemRequestFormat        = Json.format[CreateItemRequest]
-  implicit val updateItemRequestFormat        = Json.format[UpdateItemRequest]
-  implicit val partialUpdateItemRequestFormat = Json.format[PartialUpdateItemRequest]
+trait JsonSupport {
+  implicit val itemIdDecoder: JsonDecoder[ItemId] = JsonDecoder[Long].map(ItemId)
+  implicit val itemIdEncoder: JsonEncoder[ItemId] = JsonEncoder[Long].contramap(_.value)
+
+  implicit val itemDecoder: JsonDecoder[Item] = DeriveJsonDecoder.gen[Item]
+  implicit val itemEncoder: JsonEncoder[Item] = DeriveJsonEncoder.gen[Item]
+
+  implicit val dbStatusDecoder: JsonDecoder[DbStatus] = DeriveJsonDecoder.gen[DbStatus]
+  implicit val dbStatusEncoder: JsonEncoder[DbStatus] = DeriveJsonEncoder.gen[DbStatus]
+
+  implicit val createItemRequestDecoder: JsonDecoder[CreateItemRequest] = DeriveJsonDecoder.gen[CreateItemRequest]
+  implicit val createItemRequestEncoder: JsonEncoder[CreateItemRequest] = DeriveJsonEncoder.gen[CreateItemRequest]
+
+  implicit val updateItemRequestDecoder: JsonDecoder[UpdateItemRequest] = DeriveJsonDecoder.gen[UpdateItemRequest]
+  implicit val updateItemRequestEncoder: JsonEncoder[UpdateItemRequest] = DeriveJsonEncoder.gen[UpdateItemRequest]
+
+  implicit val partialUpdateItemRequestDecoder: JsonDecoder[PartialUpdateItemRequest] =
+    DeriveJsonDecoder.gen[PartialUpdateItemRequest]
+  implicit val partialUpdateItemRequestEncoder: JsonEncoder[PartialUpdateItemRequest] =
+    DeriveJsonEncoder.gen[PartialUpdateItemRequest]
+
+  implicit val emptyResponseDecoder: JsonDecoder[EmptyResponse] = DeriveJsonDecoder.gen[EmptyResponse]
+  implicit val emptyResponseEncoder: JsonEncoder[EmptyResponse] = DeriveJsonEncoder.gen[EmptyResponse]
```

Comment:
Some of these shouldn't need both Encoder and Decoder as they are only encoder or only decoded. Could you check for that? I think it wasn't possible with Play JSON.

Reply:
@jczuchnowski, you're right, I was checking and I found that a couple of these shouldn't need both Encoder and Decode.
ex: UpdateItemRequest , PartialUpdateItemRequest, EmptyResponse
I am going to refactor the code
Thanks!